credoai/artifacts/model/classification_model.py

```diff
@@ -167,15 +167,19 @@ def __init__(
             self.predict_output = check_array(
                 predict_output, ensure_2d=False, allow_nd=True
             )
+            self.__dict__["predict"] = lambda x=None: self.predict_output
+
         if predict_proba_output is not None:
             self.predict_proba_output = check_array(
                 predict_proba_output, ensure_2d=False, allow_nd=True
             )
+            self.__dict__["predict_proba"] = lambda x=None: self.predict_proba_output
+
         self.name = name
         self.tags = tags
 
-    def predict(self, X=None):
```

Comment:
Replace with a helper functions like:
```
def _wrap_array(self,  array):
      return lambda x: array

def _build_functionality(self., function_name, array):
     if array is not None:
             array = check_array(array, ensure_2d=False, allow_nd=True)
             self.__dict__[function_name] = self._wrarp_array(array)


```

Add documentation, and use those functions to define